components/shared/profile/edit/EditSocials.tsx

```diff
@@ -0,0 +1,245 @@
+                defaultValue={"@" + user?.socials?.linkedin?.username!}
+                onChange={(e) => setValue("linkedin.username", e.target.value)}
+              />
+              <Input
+                className="w-1/2 bg-black-700 text-white-300 "
+                {...register("linkedin.url")}
+                defaultValue={user.socials?.linkedin?.url!}
+                onChange={(e) => setValue("linkedin.url", e.target.value)}
+              />
+            </div>
+          </div>
+
+          <div className="flex flex-row gap-6">
+            <Image src={facebook} alt="twitter" height={32} width={32} />
+            <div className="flex h-6 flex-row gap-2 ">
+              <Input
+                {...register("facebook.username")}
+                className="w-1/2 bg-black-700 text-white-300 "
+                defaultValue={"@" + user.socials?.facebook?.username!}
+                onChange={(e) => setValue("facebook.username", e.target.value)}
+              />
+              <Input
+                {...register("facebook.url")}
+                className="w-1/2 bg-black-700 text-white-300 "
+                defaultValue={user.socials?.facebook?.url!}
+                onChange={(e) => setValue("facebook.url", e.target.value)}
+              />
+            </div>
+          </div>
+
+          <div className="flex flex-row gap-6">
+            <Image src={dribble} alt="twitter" height={32} width={32} />
+            <div className="flex h-6 flex-row gap-2 ">
+              <Input
+                {...register("dribbble.username")}
+                className="w-1/2 bg-black-700 text-white-300 "
+                defaultValue={"@" + user.socials?.dribbble?.username!}
+                onChange={(e) => setValue("dribbble.username", e.target.value)}
+              />
+              <Input
+                {...register("dribbble.url")}
+                className="w-1/2 bg-black-700 text-white-300 "
+                defaultValue={"@" + user?.socials?.dribbble?.url!}
+                onChange={
+                  (e) => setValue("dribbble.url", e.target.value) // Updated argument to "dribbble"
+                }
+              />
+            </div>
+          </div>
+
+          <div className="flex flex-row gap-6">
+            <Image src={github} alt="gh" height={32} width={32} />
+            <div className="flex h-6 flex-row gap-2 ">
+              <Input
+                {...register("github.username")}
+                defaultValue={"@" + user.socials?.github?.username!}
+                className="w-1/2 bg-black-700 text-white-300 "
+                onChange={(e) => setValue("github.username", e.target.value)}
+              />
+              <Input
+                {...register("github.url")}
+                defaultValue={user.socials?.github?.url!}
+                className="w-1/2 bg-black-700 text-white-300 "
+                onChange={(e) => setValue("github.url", e.target.value)}
+              />
```

Comment:
```suggestion
              <Input
                {...register("instagram.username")}
                defaultValue={"@" + user.socials?.instagram?.username!}
                className="w-1/2 bg-black-700 text-white-300 "
              />
              <Input
                {...register("instagram.url")}
                className="w-1/2 bg-black-700 text-white-300 "
                defaultValue={user.socials?.instagram?.url!}
              />
            </div>
          </div>

          <div className="flex flex-row gap-6">
            <Image src={linkedin} alt="twitter" height={32} width={32} />
            <div className="flex h-6 flex-row gap-2 ">
              <Input
                className="w-1/2 bg-black-700 text-white-300 "
                {...register("linkedin.username")}
                defaultValue={"@" + user?.socials?.linkedin?.username!}

              />
              <Input
                className="w-1/2 bg-black-700 text-white-300 "
                {...register("linkedin.url")}
                defaultValue={user.socials?.linkedin?.url!}

              />
            </div>
          </div>

          <div className="flex flex-row gap-6">
            <Image src={facebook} alt="twitter" height={32} width={32} />
            <div className="flex h-6 flex-row gap-2 ">
              <Input
                {...register("facebook.username")}
                className="w-1/2 bg-black-700 text-white-300 "
                defaultValue={"@" + user.socials?.facebook?.username!}
              />
              <Input
                {...register("facebook.url")}
                className="w-1/2 bg-black-700 text-white-300 "
                defaultValue={user.socials?.facebook?.url!}
              />
            </div>
          </div>

          <div className="flex flex-row gap-6">
            <Image src={dribble} alt="twitter" height={32} width={32} />
            <div className="flex h-6 flex-row gap-2 ">
              <Input
                {...register("dribbble.username")}
                className="w-1/2 bg-black-700 text-white-300 "
                defaultValue={"@" + user.socials?.dribbble?.username!}
              />
              <Input
                {...register("dribbble.url")}
                className="w-1/2 bg-black-700 text-white-300 "
                defaultValue={"@" + user?.socials?.dribbble?.url!}

              />
            </div>
          </div>

          <div className="flex flex-row gap-6">
            <Image src={github} alt="gh" height={32} width={32} />
            <div className="flex h-6 flex-row gap-2 ">
              <Input
                {...register("github.username")}
                defaultValue={"@" + user.socials?.github?.username!}
                className="w-1/2 bg-black-700 text-white-300 "
              />
              <Input
                {...register("github.url")}
                defaultValue={user.socials?.github?.url!}
                className="w-1/2 bg-black-700 text-white-300 "
              />
```

The idea behind register and react-hook-forms is that it should take care of these things for us. Try it out. It should work fine.

Reply:
I've removed all the onChange from these registered fields.